daemon/internal/newrelic/php_packages.go

```diff
@@ -7,78 +7,142 @@ package newrelic
 
 import (
 	"bytes"
+	"encoding/json"
 	"fmt"
 	"time"
-
-	"github.com/newrelic/newrelic-php-agent/daemon/internal/newrelic/log"
 )
 
 type PhpPackagesKey struct {
 	Name    string
 	Version string
 }
 
-// phpPackages represents all detected packages reported by an agent.
+// PhpPackages represents all detected packages reported by an agent
+// for a harvest cycle, as well as the filtered list of packages not
+// yet seen by the daemon for the lifecycle of the current daemon
+// process to be reported to the backend.
 type PhpPackages struct {
-	numSeen int
-	data    JSONString
+	numSeen      int
+	data         []PhpPackagesKey
+	filteredPkgs []PhpPackagesKey
 }
 
-// NumSeen returns the total number PHP packages payloads stored.
+// NumSaved returns the total number PHP packages payloads stored.
 // Should always be 0 or 1.  The agent reports all the PHP
```

Comment:
Is "Should always be 0 or 1" still valid? Now that daemon potentially holds on to multiple php packages payloads received from the agent, shouldn't this represent the number of times the daemon received php packages payload from the agent in a given harvest cycle?

Reply:
At least the comment should be updated because it is misleading.